components/clp-package-utils/clp_package_utils/scripts/native/decompress.py

```diff
@@ -26,6 +38,121 @@
 logger.addHandler(logging_console_handler)
 
 
+def get_orig_file_id(db_config: Database, path: str) -> Optional[str]:
+    """
+    :param db_config:
+    :param path: Path of the original file.
+    :return: The ID of an original file which has the given path, or None if no such file exists.
+    NOTE: Multiple original files may have the same path in which case this method returns the ID of
+    only one of them.
+    """
+    sql_adapter = SQL_Adapter(db_config)
+    with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+        db_conn.cursor(dictionary=True)
+    ) as db_cursor:
+        db_cursor.execute(
+            f"SELECT orig_file_id FROM `{CLP_METADATA_TABLE_PREFIX}files` WHERE path = (%s)",
+            (path,),
+        )
+        results = db_cursor.fetchall()
+        db_conn.commit()
+
+        if len(results) == 0:
+            logger.error("No file found for the given path.")
+            return None
+
+        if len(results) > 1:
+            logger.warning(
+                "Multiple files found for the given path."
+                " Returning the orig_file_id of one of them."
+            )
+
+        return results[0]["orig_file_id"]
+
+
+def submit_and_monitor_ir_extraction_job_in_db(
+    db_config: Database,
+    orig_file_id: str,
+    msg_ix: int,
+    target_uncompressed_size: Optional[int],
+) -> None:
+    """
+    Submits an IR extraction job to the scheduler and waits until the job finishes.
+    :param db_config:
+    :param orig_file_id:
+    :param msg_ix:
+    :param target_uncompressed_size:
+    """
+    extract_ir_config = ExtractIrJobConfig(
+        orig_file_id=orig_file_id,
+        msg_ix=msg_ix,
+        target_uncompressed_size=target_uncompressed_size,
+    )
+
+    sql_adapter = SQL_Adapter(db_config)
+    job_id = submit_query_job(sql_adapter, extract_ir_config, QueryJobType.EXTRACT_IR)
+    job_status = wait_for_query_job(sql_adapter, job_id)
+
+    if QueryJobStatus.SUCCEEDED == job_status:
+        logger.info(f"Finished IR extraction job {job_id}.")
```

Comment:
Let's return something on success/failure and then return 0/1 in `handle_ir_extraction_command`.

Reply:
same comment as do_extract.